File: Python Fundamentals/Mid Exam problems/heart_delivery.py
integers = list(map(int, input().split('@')))
current_index = 0

while True:
    command = input()
    if command == 'Love!':
        break

    given_index = int(command.split()[1])
    current_index += given_index
    if current_index > len(integers) - 1:
        current_index = 0

    if integers[current_index] != 0:
        integers[current_index] -= 2
        if integers[current_index] == 0:
            print(f"Place {current_index} has Valentine's day.")
    else:
        print(f"Place {current_index} already had Valentine's day.")

print(f"Cupid's last position was {current_index}.")

if integers.count(0) == len(integers):
    print("Mission was successful.")
else:
    print(f"Cupid has failed {len([x for x in integers if x != 0])} places.")

File: JavaScript Fundamentals/Mid Exams/Mid Exam 4/heartDelivery.js
function solve(arr) {
    let houses = arr.shift().split('@').map(x => Number(x));
    let currentIndex = 0;

    for (let command of arr) {
        if (command == 'Love!') break;
        let tokens = command.split(' ');
        if (tokens[0] == 'Jump') {
            let length = Number(tokens[1]);
            let jumpIndex = currentIndex + length;
            if (jumpIndex > houses.length - 1) currentIndex = 0;
            else currentIndex = jumpIndex;

            if (houses[currentIndex] == 0) console.log(`Place ${currentIndex} already had Valentine's day.`);
            else {
                houses[currentIndex] -= 2;
                if (houses[currentIndex] == 0) console.log(`Place ${currentIndex} has Valentine's day.`);
            }
        }
    }
    console.log(`Cupid's last position was ${currentIndex}.`);
    let failedPlaces = 0;
    for (let house of houses) {
        if (house != 0) failedPlaces++;
    }
    if (failedPlaces > 0) console.log(`Cupid has failed ${failedPlaces} places.`);
    else console.log("Mission was successful.");
}
